python/cugraph/traversal/bfs_wrapper.pyx

```diff
@@ -24,94 +24,83 @@ from libc.stdint cimport uintptr_t
 import cudf
 import numpy as np
 
-def bfs(input_graph, start, directed=True,
-        return_sp_counter=False):
+def bfs(input_graph, start, depth_limit, direction_optimizing=False):
     """
     Call bfs
     """
     # Step 1: Declare the different varibales
     cdef graph_container_t graph_container
-    # FIXME: Offsets and indices are currently hardcoded to int, but this may
-    #        not be acceptable in the future.
+
     numberTypeMap = {np.dtype("int32") : <int>numberTypeEnum.int32Type,
                      np.dtype("int64") : <int>numberTypeEnum.int64Type,
                      np.dtype("float32") : <int>numberTypeEnum.floatType,
                      np.dtype("double") : <int>numberTypeEnum.doubleType}
 
-    # Pointers required for CSR Graph
-    cdef uintptr_t c_offsets_ptr        = <uintptr_t> NULL # Pointer to the CSR offsets
-    cdef uintptr_t c_indices_ptr        = <uintptr_t> NULL # Pointer to the CSR indices
-    cdef uintptr_t c_weights = <uintptr_t>NULL
-    cdef uintptr_t c_local_verts = <uintptr_t> NULL;
-    cdef uintptr_t c_local_edges = <uintptr_t> NULL;
-    cdef uintptr_t c_local_offsets = <uintptr_t> NULL;
     weight_t = np.dtype("float32")
+    [src, dst] = graph_primtypes_wrapper.datatype_cast([input_graph.edgelist.edgelist_df['src'], input_graph.edgelist.edgelist_df['dst']], [np.int32])
+    weights = None
 
     # Pointers for SSSP / BFS
     cdef uintptr_t c_identifier_ptr     = <uintptr_t> NULL # Pointer to the DataFrame 'vertex' Series
     cdef uintptr_t c_distance_ptr       = <uintptr_t> NULL # Pointer to the DataFrame 'distance' Series
     cdef uintptr_t c_predecessor_ptr    = <uintptr_t> NULL # Pointer to the DataFrame 'predecessor' Series
-    cdef uintptr_t c_sp_counter_ptr     = <uintptr_t> NULL # Pointer to the DataFrame 'sp_counter' Series
+    if depth_limit is None:
+        depth_limit = c_bfs.INT_MAX
```

Comment:
Is this 32 bit int max?  This should be of the same type as vertices. I think this may unintentionally limit depth or crash when vertices are 64bits.

Reply:
Single gpu supports only int32 as edge_t since cudf dataframe is limited to int32 max. 
Hence we convert our vertices internally to int32 as well before passing to algorithm